crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -204,8 +205,11 @@ impl Checkpoint {
         self.stored.sequence_number as u64
     }
 
-    pub(crate) fn network_total_transactions_impl(&self) -> u64 {
-        self.stored.network_total_transactions as u64
+    pub(crate) fn network_total_transactions_impl(&self) -> Result<u64, Error> {
+        let total_tx = self.stored.max_tx_sequence_number.ok_or(Error::Internal(
+            "Checkpoint max_tx_sequence_number should not be null".to_string(),
+        ))?;
+        Ok(total_tx as u64)
     }
```

Comment:
I thought the nullable move was for migrations and scheme changes, and on the reader end, if it's a in-fact-non-null column, we should fail fast and error whenever we found a null?

Reply:
> I thought the nullable move was for migrations and scheme changes

It is, in the initial instance, but also note my earlier comment:

> We do try and make fields in the GraphQL schema nullable/optional if we think they might go away because of deprecations, but because we haven't been forced to do that by the types, we haven't done it everywhere, so making the columns nullable will make sure we do that thoroughly.

This will not be possible if we insulate the GraphQL schema from the nullability of the DB schema by failing fast, and then re-wrapping it in an `Option` (the types will not be forcing us in that case), and in the end, if we do deprecate a field, we would then need to remember to undo the unwrap/rewrap to prevent the deprecation from breaking existing queries because of the fail fast logic.